src/EditorFeatures/TestUtilities/NavigateTo/AbstractNavigateToTests.cs

```diff
@@ -74,23 +85,30 @@ public abstract class AbstractNavigateToTests
             }
         }
 
-        protected TestWorkspace SetupWorkspace(XElement workspaceElement)
+        private protected TestWorkspace SetupWorkspace(
+            XElement workspaceElement, ExportProvider exportProvider, 
+            Func<TestWorkspace, IDocumentTrackingService> createTrackingService)
         {
-            var workspace = TestWorkspace.Create(workspaceElement, exportProvider: TestExportProvider.ExportProviderWithCSharpAndVisualBasic);
-            InitializeWorkspace(workspace);
+            var workspace = TestWorkspace.Create(workspaceElement, exportProvider: exportProvider);
```

Comment:
>exportProvider: [](start = 67, length = 15)

nit: consider not naming this argument (if possible)

Reply:
@sharwell is this the way you imagine people inject Mock service to workspace? or some other way? I know you said, in test, we should never use mock object. so not sure whether using lock like this is okay or not.